frontend/src/components/Login/Signup.tsx

```diff
@@ -31,15 +33,19 @@ export function Signup() {
     setUserCredentials(currentuserCredentials => ({ ...currentuserCredentials, [name]: value }));
   };
 
-  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+  const OnSignUp = async (e: React.FormEvent<HTMLFormElement>) => {
     e.preventDefault();
-    try {
-      const res = await signUp(userCredentials.email, userCredentials.password);
-      if (res) navigate('/home');
-    } catch (err) {
-      if (err instanceof FirebaseError) {
-        setError(err.message);
-      }
+    const res = await signUpApi(
```

Comment:
Why did you remove the `try/catch` block?

Whenever you are dealing with a `Promise`, you need to handle the case where the `Promise` rejects with an error.

This means using `.catch` if you are using `.then`, or using a `try`/`catch` block in case of `async`/`await`.

Bring the block back.